masonry/src/widget/board.rs

```diff
@@ -0,0 +1,633 @@
+// Copyright 2018 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A widget that arranges its children in a one-dimensional array.
+
+use accesskit::Role;
+use smallvec::SmallVec;
+use tracing::{trace_span, warn, Span};
+use vello::kurbo::{
+    self, Affine, Arc, BezPath, Circle, CircleSegment, CubicBez, Ellipse, Line, PathEl, PathSeg,
+    QuadBez, RoundedRect, Shape as _, Stroke, Vec2,
+};
+use vello::peniko::{Brush, Fill};
+use vello::Scene;
+
+use crate::widget::WidgetMut;
+use crate::{
+    AccessCtx, AccessEvent, BoxConstraints, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx, PaintCtx,
+    Point, PointerEvent, Rect, Size, StatusChange, TextEvent, Widget, WidgetId, WidgetPod,
+};
+
+/// A container with absolute positioning layout.
+pub struct Board {
+    children: Vec<Child>,
+    origin: Point,
+    scale: Vec2,
+}
+
+/// Parameters for an item in a [`Board`] container.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct BoardParams {
+    origin: Point,
+    size: Size,
+}
+
+pub struct Shape {
+    shape: KurboShape,
+    transform: Affine,
+    fill_style: Fill,
+    fill_brush: Brush,
+    fill_brush_transform: Option<Affine>,
+    stroke_style: Stroke,
+    stroke_brush: Brush,
+    stroke_brush_transform: Option<Affine>,
+}
```

Comment:
This (and the impl blocks) could probably be moved to a separate file.

I suggest `StaticShape` or `DrawnShape` as a name to differentiate from the `Shape` trait.

Reply:
KurboShape the widget seems more reasonable.